### DISCUSSION TITLE
RuntimeError: probability tensor contains either `inf`, `nan` or element < 0..or images are completely black

### DISCUSSION THREAD

**Discussion by asahii** [lllyasviel/Fooocus#2736] [Q&A]
I keep getting the error: RuntimeError: probability tensor contains either inf, nan or element < 0. If I turn off style-Fooocus V2, the error disappears, but the generated images are completely black.

E:\fooocus>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.3.1
[Cleanup] Attempting to delete content of temp dir C:\Users\mi\AppData\Local\Temp\fooocus
[Cleanup] Cleanup successful
Total VRAM 8192 MB, total RAM 32717 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce RTX 2080 : native
VAE dtype: torch.float32
Using pytorch cross attention
Refiner unloaded.
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.text_projection'}
Base model loaded: E:\fooocus\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors
Request to load LoRAs [['sd_xl_offset_example-lora_1.0.safetensors', 0.1], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [E:\fooocus\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors].
Loaded LoRA [E:\fooocus\Fooocus\models\loras\sd_xl_offset_example-lora_1.0.safetensors] for UNet [E:\fooocus\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors] with 788 keys at weight 0.1.
Fooocus V2 Expansion: Vocab with 642 words.
Fooocus Expansion engine loaded for cuda:0, use_fp16 = True.
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
[Fooocus Model Management] Moving model(s) has taken 0.56 seconds
Started worker with PID 4184
App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865
[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ControlNet Softness = 0.25
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 4.0
[Parameters] Seed = 8778731334797361813
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
Traceback (most recent call last):
  File "E:\fooocus\Fooocus\modules\async_worker.py", line 913, in worker
    handler(task)
  File "E:\fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\fooocus\Fooocus\modules\async_worker.py", line 486, in handler
    expansion = pipeline.final_expansion(t['task_prompt'], t['task_seed'])
  File "E:\fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\fooocus\Fooocus\extras\expansion.py", line 120, in __call__
    features = self.model.generate(**tokenized_kwargs,
  File "E:\fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\fooocus\python_embeded\lib\site-packages\transformers\generation\utils.py", line 1572, in generate
    return self.sample(
  File "E:\fooocus\python_embeded\lib\site-packages\transformers\generation\utils.py", line 2655, in sample
    next_tokens = torch.multinomial(probs, num_samples=1).squeeze(1)
RuntimeError: probability tensor contains either `inf`, `nan` or element < 0
Total time: 1.38 seconds

off style-Fooocus V2, the error disappears, but the generated images are completely black
Total time: 1.38 seconds
[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ControlNet Softness = 0.25
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 4.0
[Parameters] Seed = 7758933833899865753
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
[Fooocus] Processing prompts ...
[Fooocus] Encoding positive #1 ...
[Fooocus Model Management] Moving model(s) has taken 0.15 seconds
[Fooocus] Encoding positive #2 ...
[Fooocus] Encoding negative #1 ...
[Fooocus] Encoding negative #2 ...
[Parameters] Denoising Strength = 1.0
[Parameters] Initial Latent shape: Image Space (896, 1152)
Preparation time: 4.92 seconds
[Sampler] refiner_swap_method = joint
[Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828
Requested to load SDXL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 2.39 seconds
100%|██████████████████████████████████████████████████████████████████████████████████| 30/30 [00:16<00:00,  1.86it/s]
Requested to load AutoencoderKL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.21 seconds
Image generated with private log at: E:\fooocus\Fooocus\outputs\2024-04-09\log.html
Generating and saving time: 19.97 seconds
[Sampler] refiner_swap_method = joint
[Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828
Requested to load SDXL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 1.47 seconds

**Comment by asahii** [comment]
![Uploading %7BE67DA582-8F4A-4709-AAAD-B7E91FE2FA65%7D.png…]()

**Comment by mashb1t** [comment]
Please try to add the following arguments: `--vae-in-fp16 --unet-in-fp16 --all-in-fp16`
If this doesn't help, then try everything in fp32.
Doing so allows us to debug further if one element isn't compatible with your hardware and/or if it is an issue with fp operations.

Also see https://github.com/lllyasviel/Fooocus/pull/1710#issuecomment-2004640757

**Reply by asahii** [reply]
E:\fooocus>.\python_embeded\python.exe -s Fooocus\entry_with_update.py --vae-in-fp16 --unet-in-fp16 --all-in-fp16
Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py', '--vae-in-fp16', '--unet-in-fp16', '--all-in-fp16']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.3.1
[Cleanup] Attempting to delete content of temp dir C:\Users\mi\AppData\Local\Temp\fooocus
[Cleanup] Cleanup successful
Total VRAM 8192 MB, total RAM 32717 MB
Forcing FP16.
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce RTX 2080 : native
VAE dtype: torch.float16
Using pytorch cross attention
Refiner unloaded.
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids'}
Base model loaded: E:\fooocus\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors
Request to load LoRAs [['sd_xl_offset_example-lora_1.0.safetensors', 0.1], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [E:\fooocus\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors].
Loaded LoRA [E:\fooocus\Fooocus\models\loras\sd_xl_offset_example-lora_1.0.safetensors] for UNet [E:\fooocus\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors] with 788 keys at weight 0.1.
Fooocus V2 Expansion: Vocab with 642 words.
Fooocus Expansion engine loaded for cuda:0, use_fp16 = True.
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
[Fooocus Model Management] Moving model(s) has taken 0.44 seconds
Started worker with PID 14152
App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865
[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ControlNet Softness = 0.25
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 4.0
[Parameters] Seed = 3051205396359684486
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
Traceback (most recent call last):
  File "E:\fooocus\Fooocus\modules\async_worker.py", line 913, in worker
    handler(task)
  File "E:\fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\fooocus\Fooocus\modules\async_worker.py", line 486, in handler
    expansion = pipeline.final_expansion(t['task_prompt'], t['task_seed'])
  File "E:\fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\fooocus\Fooocus\extras\expansion.py", line 120, in __call__
    features = self.model.generate(**tokenized_kwargs,
  File "E:\fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\fooocus\python_embeded\lib\site-packages\transformers\generation\utils.py", line 1572, in generate
    return self.sample(
  File "E:\fooocus\python_embeded\lib\site-packages\transformers\generation\utils.py", line 2655, in sample
    next_tokens = torch.multinomial(probs, num_samples=1).squeeze(1)
RuntimeError: probability tensor contains either `inf`, `nan` or element < 0
Total time: 6.37 seconds

**Comment by asahii** [comment]
.\python_embeded\python.exe -s Fooocus\entry_with_update.py --all-in-fp32
pause

I can run it by adding this parameter, what's the problem?


E:\fooocus>.\python_embeded\python.exe -s Fooocus\entry_with_update.py --all-in-fp32
Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py', '--all-in-fp32']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.3.1
[Cleanup] Attempting to delete content of temp dir C:\Users\mi\AppData\Local\Temp\fooocus
[Cleanup] Cleanup successful
Total VRAM 8192 MB, total RAM 32717 MB
Forcing FP32, if this improves things please report it.
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce RTX 2080 : native
VAE dtype: torch.float32
Using pytorch cross attention
Refiner unloaded.
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_l.text_projection'}
Base model loaded: E:\fooocus\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors
Request to load LoRAs [['sd_xl_offset_example-lora_1.0.safetensors', 0.1], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [E:\fooocus\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors].
Loaded LoRA [E:\fooocus\Fooocus\models\loras\sd_xl_offset_example-lora_1.0.safetensors] for UNet [E:\fooocus\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors] with 788 keys at weight 0.1.
Fooocus V2 Expansion: Vocab with 642 words.
Fooocus Expansion engine loaded for cpu, use_fp16 = False.
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
Started worker with PID 1392
App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865
[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ControlNet Softness = 0.25
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 4.0
[Parameters] Seed = 4516206882057630634
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
[Prompt Expansion] 1girl, altered color, detailed gorgeous, amazing, elegant, intricate, highly enhanced, sharp focus, professional, cinematic, unique, best, attractive, cute, charming, pretty, perfect, focused, colorful, cool, loving, relaxed, friendly, warm, lovely, wonderful, futuristic, iconic, inspiring, beautiful, dramatic, background, illuminated, light, shining
[Fooocus] Preparing Fooocus text #2 ...
[Prompt Expansion] 1girl, intricate, elegant, highly detailed, wonderful quality, serious detail, cool color, dramatic light, professional, charming, pretty, best, full, true, whole, perfect, pure, cheerful, cute, innocent, majestic, beautiful, novel, epic, background, colorful, illuminated, very artistic, sharp focus, enhanced, complex, vibrant, symmetry, fine
[Fooocus] Encoding positive #1 ...
[Fooocus] Encoding positive #2 ...
[Fooocus] Encoding negative #1 ...
[Fooocus] Encoding negative #2 ...
[Parameters] Denoising Strength = 1.0
[Parameters] Initial Latent shape: Image Space (896, 1152)
Preparation time: 23.45 seconds
[Sampler] refiner_swap_method = joint
[Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828
Requested to load SDXL
Loading 1 new model
loading in lowvram mode 4667.737003326416
[Fooocus Model Management] Moving model(s) has taken 17.54 seconds
 73%|████████████████████████████████████████████████████████████▏                     | 22/30 [01:13<00:26,  3.26s/it]

**Reply by Rogala** [reply]
[Reinstall the driver](https://www.nvidia.com/download/driverResults.aspx/223954/en-us/)
